### PR TITLE
[opentelemetry] Test OTLP log ingestion in Agent

### PR DESCRIPTION
**Pull request by songy23**
## Description

Test OTLP log ingestion in Agent, also enable some additional checks on metrics.

## Motivation

This covers the E2E test on the new feature in Agent which accepts OTLP logs.

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/otel_tracing_e2e/test_e2e.py [hunk 1]
```diff
@@ -120,24 +119,42 @@ class Test_OTelLogE2E:
     def setup_main(self):
         self.r = weblog.get(path="/basic/log")
         self.use_128_bits_trace_id = False
+        time.sleep(5)  # wait a bit for logs agent to submit logs
```
Comment (cbeauchesne) [inline tests/otel_tracing_e2e/test_e2e.py:122]: You must achieve this by adding a argument to this line : 

https://github.com/DataDog/system-tests/blob/d0e37b5accdc81975e22564d83a23e46e2a3411e/utils/_context/_scenarios.py#L1102

The argument is `backend_interface_timeout`, it's 0 by default, so I guess you want at least 20. You must also remove any sleep in `setup` methods (sorry, I miss that point in a preview PR review ...)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -533,7 +533,7 @@ jobs:
       if: contains(github.event.pull_request.labels.*.name, 'build-proxy-image')
       run: ./build.sh -i proxy
     - name: Build agent
-      run: SYSTEM_TEST_BUILD_ATTEMPTS=3 ./build.sh java_otel -i agent
+      run: SYSTEM_TEST_BUILD_ATTEMPTS=3 AGENT_BASE_IMAGE="datadog/agent:7.48.0-rc.3" ./build.sh java_otel -i agent
```
Comment (cbeauchesne) [inline .github/workflows/ci.yml:536]: Can you use the existing way ? 

`echo "datadog/agent:7.48.0-rc.3" > binaries/agent-image`

File: utils/build/build.sh [hunk 3]
```diff
@@ -165,10 +165,12 @@ build() {
                 .
 
         elif [[ $IMAGE_NAME == agent ]]; then
-            if [ -f ./binaries/agent-image ]; then
-                AGENT_BASE_IMAGE=$(cat ./binaries/agent-image)
-            else
-                AGENT_BASE_IMAGE="datadog/agent"
+            if [[ -z "${AGENT_BASE_IMAGE:-}" ]]; then
```
Comment (cbeauchesne) [inline utils/build/build.sh:168]: No need to add this (see comment above)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (cbeauchesne) [inline .github/workflows/ci.yml:536]: Just for my understanding, why hard coding this version ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (songy23) [reply .github/workflows/ci.yml]: 👍 Looks better

Comment (songy23) [reply .github/workflows/ci.yml]: The new feature that we want to test is only available in 7.48. If I don't specify the `rc.3` version, Docker will pull 7.47 by default.

File: tests/otel_tracing_e2e/test_e2e.py (hunk 1 above)
Comment (songy23) [reply tests/otel_tracing_e2e/test_e2e.py]: Makes sense, used it instead

File: utils/build/build.sh (hunk 3 above)
Comment (songy23) [reply utils/build/build.sh]: Reverted

File: .github/workflows/ci.yml (hunk 2 above)
Comment (songy23) [reply .github/workflows/ci.yml]: Great, thanks!

Comment (cbeauchesne) [reply .github/workflows/ci.yml]: -> #1577

Comment (cbeauchesne) [reply .github/workflows/ci.yml]: Merged. You can update your branch, remove the hardcoded agent version in the CI, and then use `@released(agent="1.2")` on test classes, `1.2` being the version where the feature is supposed to be ok.

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
CI is failing

**Comment by songy23** [comment]
@cbeauchesne should be good now, the failed Python tests don't seem relevant.
```
   8 | >>> RUN /binaries/install_ddtrace.sh
   9 |     
  10 |     ENV DD_TRACE_HEADER_TAGS='user-agent:http.request.headers.user-agent'
--------------------
ERROR: failed to solve: process "/bin/sh -c /binaries/install_ddtrace.sh" did not complete successfully: exit code: 1
Build step failed after 3 attempts
```

**Pull request merged**